TestErrorSpam: Restarting the docker service may improve performance

as seen in 
https://storage.googleapis.com/minikube-builds/logs/7586/d512c84/Docker_Linux.html#fail_TestErrorSpam

```
=== RUN   TestErrorSpam
=== PAUSE TestErrorSpam
=== CONT  TestErrorSpam
--- FAIL: TestErrorSpam (168.88s)
error_spam_test.go:42: (dbg) Run:  out/minikube-linux-amd64 start -p nospam-20200410T153613.393267403-15978 -n=1 --memory=2250 --wait=false --driver=docker 
error_spam_test.go:42: (dbg) Done: out/minikube-linux-amd64 start -p nospam-20200410T153613.393267403-15978 -n=1 --memory=2250 --wait=false --driver=docker : (2m15.474942738s)
error_spam_test.go:57: unexpected stderr line: "* Restarting the docker service may improve performance."
error_spam_test.go:57: unexpected stderr line: "stdout:"
error_spam_test.go:57: unexpected stderr line: "serviceaccount/storage-provisioner unchanged"
error_spam_test.go:57: unexpected stderr line: "clusterrolebinding.rbac.authorization.k8s.io/storage-provisioner unchanged"
error_spam_test.go:57: unexpected stderr line: "stderr:"
error_spam_test.go:49: unexpected error log in stderr: "Error from server (ServerTimeout): error when creating \"/etc/kubernetes/addons/storage-provisioner.yaml\": No API token found for service account \"storage-provisioner\", retry after the token is automatically created and added to the service account"
error_spam_test.go:57: unexpected stderr line: "]"
error_spam_test.go:69: *** TestErrorSpam FAILED at 2020-04-10 15:38:28.868492686 -0700 PDT m=+725.487531520
helpers.go:198: (dbg) Run:  out/minikube-linux-amd64 status --format={{.Host}} -p nospam-20200410T153613.393267403-15978
helpers.go:198: (dbg) Done: out/minikube-linux-amd64 status --format={{.Host}} -p nospam-20200410T153613.393267403-15978: (1.579189399s)
```

--- a/test/integration/error_spam_test.go
+++ b/test/integration/error_spam_test.go
@@ -25,6 +25,17 @@ import (
 	"testing"
 )
 
+// stderrWhitelist are regular expressions acceptable to find in normal stderr
+var stderrWhitelist = []string{
+	// kubectl out of date warning
+	`kubectl`,
+	// slow docker warning
+	`slow|long time|Restarting the docker service may improve`,
+}
+
+// stderrWhitelistRe combines rootCauses into a single regex
+var stderrWhitelistRe = regexp.MustCompile(strings.Join(stderrWhitelist, "|"))
+
 // TestErrorSpam asserts that there are no errors displayed
 func TestErrorSpam(t *testing.T) {
 	if NoneDriver() {
@@ -41,20 +52,22 @@ func TestErrorSpam(t *testing.T) {
 
 	rr, err := Run(t, exec.CommandContext(ctx, Target(), args...))
 	if err != nil {
-		t.Errorf("failed to start minikube with args: %q : %v", rr.Command(), err)
+		t.Errorf("%q failed: %v", rr.Command(), err)
 	}
 
 	for _, line := range strings.Split(rr.Stderr.String(), "\n") {
 		if strings.HasPrefix(line, "E") {
-			t.Errorf("unexpected error log in stderr: %q", line)
+			t.Errorf("unexpected error log: %q", line)
 			continue
 		}
 
-		if strings.Contains(line, "kubectl") || strings.Contains(line, "slow") || strings.Contains(line, "long time") {
+		if stderrWhitelistRe.MatchString(line) {
+			t.Logf("acceptable stderr: %q", line)
 			continue
 		}
+
 		if len(strings.TrimSpace(line)) > 0 {
-			t.Errorf("unexpected stderr line: %q", line)
+			t.Errorf("unexpected stderr: %q", line)
 		}
 	}
 
@@ -62,7 +75,7 @@ func TestErrorSpam(t *testing.T) {
 		keywords := []string{"error", "fail", "warning", "conflict"}
 		for _, keyword := range keywords {
 			if strings.Contains(line, keyword) {
-				t.Errorf("unexpected %q in stdout line: %q", keyword, line)
+				t.Errorf("unexpected %q in stdout: %q", keyword, line)
 			}
 		}
 	}

--- a/test/integration/error_spam_test.go
+++ b/test/integration/error_spam_test.go
@@ -21,6 +21,7 @@ package integration
 import (
 	"context"
 	"os/exec"
+	"regexp"
 	"strings"
 	"testing"
 )

